### ISSUE TITLE
Pydantic dependencies disappeared between v2.6.4 to v2.7.0

### ISSUE THREAD

**Issue by roismauro** [pydantic/pydantic#9249]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

I'm using poetry; when using pydantic v2.6.4, locking brings its dependencies but after updating to v2.7.0 it doesn't.
I can't even run my tests because pydantic-core is missing:

./../.venv/lib/python3.8/site-packages/pydantic/__init__.py:389: in __getattr__
    module = import_module(module_name, package=package)
../../.venv/lib/python3.8/site-packages/pydantic/main.py:12: in <module>
    import pydantic_core
E   ModuleNotFoundError: No module named 'pydantic_core'

poetry show results:

poetry show pydantic
 name         : pydantic                                
 version      : 2.6.4                                   
 description  : Data validation using Python type hints 

dependencies
 - annotated-types >=0.4.0
 - pydantic-core 2.16.3
 - typing-extensions >=4.6.1

poetry show pydantic
 name         : pydantic 
 version      : 2.7.0    
 description  :          

Workaround is explicitly adding pydantic dependencies:
pydantic-core = "^2.0.0"
typing-extensions = "*"
annotated-types = "*"

### Python, Pydantic & OS Version

```Text
python -c "import pydantic.version; print(pydantic.version.version_info())"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "***/.venv/lib/python3.8/site-packages/pydantic/version.py", line 26, in version_info
    import pydantic_core._pydantic_core as pdc
ModuleNotFoundError: No module named 'pydantic_core'
```

**Comment by user-name-name** [comment]
I met the same issue after update `black` today =(

**Comment by enkidulan** [comment]
same here

**Comment by Viicos** [comment]
By any chance, are you using AWS CodeArtifact or any other custom package repository?

Could you provide your Poetry version?

**Comment by Viicos** [comment]
This is a Poetry issue, see https://github.com/python-poetry/poetry/issues/9301 for related tickets and a TL;DR including a fix.

**Comment by enkidulan** [comment]
Upgrading `poetry` to `1.8.2` didn't solve the issue for me :(

**Comment by roismauro** [comment]
> This is a Poetry issue, see [python-poetry/poetry#9301](https://github.com/python-poetry/poetry/issues/9301) for related tickets and a TL;DR including a fix.

I'm using poetry v1.8.2, jfrog cloud artifactory but having the issue

**Comment by Viicos** [comment]
Please follow the workarounds provided [here](https://github.com/python-poetry/poetry/issues/9294#issuecomment-2048116196).

**Comment by roismauro** [comment]
> Please follow the workarounds provided [here](https://github.com/python-poetry/poetry/issues/9294#issuecomment-2048116196).

Tried it but the dependencies are still missing

**Comment by Viicos** [comment]
> Tried it but the dependencies are still missing

Then you probably need to open an issue @ Poetry, but you will probably have to provide additional info on what you've tried as there's already a bunch of issues raised about this problem

**Comment by sydney-runkle** [comment]
Marking as resolved, thanks @Viicos!

**Comment by Philaeux** [comment]
TLDR of commands to run:

- update pkginfo pip pipx `python -m pip install --upgrade pip pipx pkginfo`
- update poetry `pipx upgrade poetry`
- clear cache `poetry cache clear --all .`
- rebuild graph of dependencies `poetry update`
